x-pack/legacy/plugins/infra/public/containers/logs/log_summary/api/log_summary.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { fold } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { identity } from 'fp-ts/lib/function';
+import { kfetch } from 'ui/kfetch';
+
+import { throwErrors, createPlainError } from '../../../../../common/runtime_types';
+
+import {
+  LOGS_SUMMARY_PATH,
+  LOGS_SUMMARY_HIGHLIGHTS_PATH,
+  LogsSummaryRequest,
+  logsSummaryRequestRT,
+  logsSummaryResponseRT,
+  LogsSummaryHighlightsRequest,
+  logsSummaryHighlightsRequestRT,
+  logsSummaryHighlightsResponseRT,
+} from '../../../../../common/http_api';
+
+export const fetchLogSummary = async (requestArgs: LogsSummaryRequest) => {
+  const response = await kfetch({
+    method: 'POST',
+    pathname: LOGS_SUMMARY_PATH,
+    body: JSON.stringify(logsSummaryRequestRT.encode(requestArgs)),
+  });
+
+  return pipe(
+    logsSummaryResponseRT.decode(response),
+    fold(throwErrors(createPlainError), identity)
+  );
+};
+
+export const fetchLogSummaryHighlights = async (requestArgs: LogsSummaryHighlightsRequest) => {
```

Comment:
Would it make sense to split them into two files such as `get_log_summary.ts` and `get_log_summary_highlights.ts`?

That would scale better for further log summary API additions. (I could imagine adding `log_summary_anomalies.ts` or `log_summary_user_comments.ts`.)

Reply:
I have no strong opinion about it. The reasoning to have it on the same file was the same as with the URL, treating `highlights` as a specialised version of `summary`.

For future-proofing, I might misunderstand how we are modeling logs. In my head we have:

- Log entries, and log _entries_ summary.
- Log anomalies, and log _anomalies_ summary.
- etc.

Meaning, the summary is a specialised view of some other entity (entries, anomalies, etc) and not the other way around.

After writing this out loud, I think `logs` might be too generic as a root. I could structure it like this:

```
.
└── log_entries
    ├── entries.ts # In the near future
    ├── summary.ts
    └── summary_highlights.ts
```

What do you think?

Reply:
My current mental model was that we have two views on the logs: the individual entries and the summary. Over time both have been incrementally extended to be annotated by highlights. Projecting into the future either or both might be extended to display anomalies, comments and such.

The only arguments I can think of nesting the hierarchy one way over the other right now would be

* the entries and the summary might not always gain the features symmetrically (e.g. log rate anomalies might be shown in the summary but not in the text).
* anomalies is not a particularly well-defined concept for us at the moment.

My original suggestion of splitting the functions into files was aimed exactly at avoiding any such nesting decision by keeping the hierarchy flat.

Reply:
I think we agree :)

Let me add a commit to be sure that we actually agree though :D